### PR TITLE
Support Petri Nets and Syntax Trees

### PR DESCRIPTION
**Pull request by MaisiKoleni**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] #2168 got merged.

### Motivation and Context
Apollon supports petri nets and syntax trees with the new version `2.2.0`, introduced in #2168.

We now want to add support for both types in Artemis as well.

### Description
- Added syntax tree and petri nets to the Artemis `DiagramType`s
- Added models for both types for the compass subsystem and added them to the compass JSON parser.
- Added tests for those two compass models
- [x] _TODO: move JSON from text blocks into files, depends a bit on #2122._

### Steps for Testing

0. Review the Code, at least the most important classes.

_**Do this for both petri nets and syntax trees:**_
1. Create a modelling exercise for the new diagram type as instructor _(preferred: with start, due and assessment due date)_.
2. Check that the example solution and example submission can be created 
    (available when editing the newly created exercise)
3. Participate as a student, change your submission, within the due date. 
    Everything should work as expected, no lost changes, no errors, ...
    Play around with the diagram editor a bit as well.
4. Wait for due date to pass or set it
5. Logged in as tutor, go to the tutor exercise dashboard for your new exercise.
6. You should see the grading instructions and the example solution.
7. You should then have to work on the example submission(s) first.
8. Now you can assess the submission done earlier.
    Check that assessment works and all elements and connections can be assessed with points and explanations.
9. As a student again, check your submission. You should not see the grading yet.
11. As instructor, check the exercise scores table (over course management) and take a look at the submission.
     You should see the submission and assessment fully and correctly displayed.
10. Wait for the assessment due date or set it.
11. Logged in as student, check your score and look at the assessment.
     You should now see everything the instructor could see (apart from the assessor/tutor).
12. I would be good if you have/generate more submissions with different similarity for the next step.
13. Over course management, go to view exercise and then on check plagiarism.
14. You should now see some submissions compared to each other if they are similar. 
      Check that the reported similarity roughly matches your perceived similarity.

### Test Coverage

- `de.tum.in.www1.artemis.service.compass.umlmodel.petrinet`  - 83%
- `de.tum.in.www1.artemis.service.compass.umlmodel.syntaxtree` - 80%
- `de.tum.in.www1.artemis.service.compass.controller` - 95%

#### Syntax Tree
image

image

#### Petri Net
image

image

### CODE REVIEW

**Review by krusche — DISMISSED** [review]

File: src/main/java/de/tum/in/www1/artemis/service/compass/controller/UMLModelParser.java [hunk 1]
```diff
@@ -813,6 +814,161 @@ private static UMLControlFlow parseControlFlow(JsonObject controlFlowJson, Map<S
...
+                }
+                default -> {
+                    // ignore unknown elements
+                }
+            }
+        }
+
+        // loop over all JSON control flow elements and create syntax tree links
+        for (JsonElement rel : relationships) {
+            Optional<PetriNetArc> useCaseAssociation = parsePetriNetArc(rel.getAsJsonObject(), allElementsMap);
```
Comment (TobiasPr) [inline src/main/java/de/tum/in/www1/artemis/service/compass/controller/UMLModelParser.java:857]: weird name for petri net arc

File: src/main/java/de/tum/in/www1/artemis/service/compass/controller/UMLModelParser.java [hunk 2]
```diff
@@ -813,6 +814,161 @@ private static UMLControlFlow parseControlFlow(JsonObject controlFlowJson, Map<S
...
+                }
+                default -> {
+                    // ignore unknown elements
+                }
+            }
+        }
+
+        // loop over all JSON control flow elements and create syntax tree links
+        for (JsonElement rel : relationships) {
+            Optional<SyntaxTreeLink> useCaseAssociation = parseSyntaxTreeLink(rel.getAsJsonObject(), allElementsMap);
```
Comment (TobiasPr) [inline src/main/java/de/tum/in/www1/artemis/service/compass/controller/UMLModelParser.java:936]: same here, shouldn't `useCaseAssociation` be something like `syntacTreeLink`?

**Review by TobiasPr — COMMENTED** [review]
just had a quick peek, looks good - had two question to reference namings, see comments

**Review by krusche — APPROVED** [review]
Did not test this, but code looks good 👍

**Review by kloessst — APPROVED** [review]
Worked fine in the testing session. Looked briefly over the code and nothing caught my attention. Well done.

**Review by fde312 — APPROVED** [review]
Tested it in the test session and code looks good as far as I can tell. Good job! 🚀

**Pull request merged**